src/main/java/com/manoelcampos/javadoc/coverage/configuration/Option.java

```diff
@@ -0,0 +1,32 @@
+package com.manoelcampos.javadoc.coverage.configuration;
+
+import lombok.*;
+
+@ToString
+@Getter
+@AllArgsConstructor
+abstract class Option<T> {
```

Comment:
I was thinking that it would be better to use [Apache Commons Cli](https://commons.apache.org/proper/commons-cli/usage.html) to parse command line arguments. But nevermind if it's to much work for you.

Reply:
I looked into it briefly before writing my own option class. What we need are generic default values, and I didn't see this capability when looking at the usage examples there